core/dbt/events/types.proto

```diff
@@ -1289,6 +1289,85 @@ message NoNodesSelected {
     EventInfo info = 1;
 }
 
+// Q039
+message DebugEnvironmentDetails {
+    EventInfo info = 1;
+    string msg = 2;
```

Comment:
All of these are `msg` because that's really all I'm passing to the exceptions: various messages. The messages are so flexible that trying to define messages in `def message` functions felt "wrong" on gut reaction. Happy to adjust!